Audio feature popovers

zacharygilmour17 opened the pull request:
<!-- This is a template - do add to or remove from as needed -->
Some changes to featuremenu component + information icon included
<!-- Please provide a general summary of your changes in the Title above 🚀 -->
## Context
from discord

## Description 💬
2) Add an info icon to the top right and explain what the audio features are

zacharygilmour17 committed:
testing changes

zacharygilmour17 committed:
changed padding so audiofeatures were smaller

zacharygilmour17 committed:
spacing

CarlinWilliamson commented on src/components/FeatureSelector/FeatureMenu.tsx:
```diff
@@ -1,35 +1,86 @@
+                  }}
+                  disableRestoreFocus
+                 >
+                {popupAnchor == null ? "" : "These features make up a song"}
```
This text doesn't really tell the user much...

Maybe this popup should be on the text in each slider. Give a one liner as to what each feature means there

Also you don't want to do this `popupAnchor == null` check. It makes the popover look weird when it's closing

zacharygilmour17 replied:
removed check and changed location to audioFeatureSlider

CarlinWilliamson commented on src/components/FeatureSelector/FeatureMenu.tsx:
```diff
@@ -1,35 +1,86 @@
-          {Options.filter(o => !props.hidden.includes(o.name)).map((o, index) => (<MenuItem key={index} onClick={() => props.onSelect(o)}>{o.name}</MenuItem>))}
-        </Menu>
+      <div className = {classes.root}>
+        <List component={Paper} disablePadding={true}>
```
I'm not sure about this paper bit. Nowhere else do we put a paper inside a paper.

It's consuming more screenspace and I'm not sure what the benefit is

zacharygilmour17 replied:
removed list altogether

CarlinWilliamson reviewed (changes requested):
Sorry my task description on discord wasn't great. See my comments

zacharygilmour17 committed:
change featureicons and added slider info

- info now appears on sliders
- info removed from audio features menu
- list removed from audiofeaturesmenu

CarlinWilliamson commented on src/components/FeatureSelector/AudioFeatureSlider.tsx:
```diff
@@ -21,11 +29,39 @@ export default function (props: {
+          onMouseLeave={() => setPopupAnchor(null)}
+        />
+      </Box>
+      <Popover
```
Could the popover go on the text instead of adding an icon.

IMO there is already a lot going on there with the name, slider and delete button

CarlinWilliamson commented on src/components/FeatureSelector/Defaults.tsx:
```diff
@@ -26,7 +28,8 @@ const options: FeatureSliderData[] = [
     currentMax: 100,
-    units: '%'
+    units: '%',
+    description : 'Energy represents a perceptual measure of intensity and activity.'
```
```suggestion
    description : 'Energy describes intensity and activity of a track.'
```
wtf is a perceptual measure??

CarlinWilliamson commented on src/components/FeatureSelector/Defaults.tsx:
```diff
@@ -35,7 +38,8 @@ const options: FeatureSliderData[] = [
     currentMax: 100,
-    units: '%'
+    units: '%',
+    description: 'Instrumentalness predicts whether a track contains no vocals.'
```
It looks like this feature is inverted
i.e. 100% intrumentalness means there are vocals...

Could we rename it or invert the values in parsers.tsx

zacharygilmour17 replied:
it's not inverted its just a confidence rating from spotify
beethoven gave me 80 instrumentalness
rap songs gave me 1 instrumentalness

CarlinWilliamson replied:
so the documentation is wrong? ok...

CarlinWilliamson commented on src/components/FeatureSelector/Defaults.tsx:
```diff
@@ -53,7 +58,8 @@ const options: FeatureSliderData[] = [
     currentMax: 100,
-    units: '%'
+    units: '%',
+    description : 'Speechiness detects the presence of spoken words in a track'
```
```suggestion
    description : 'Speechiness detects the presence of spoken (not sung) words in a track'
```

I don't think this is clear enough. The documentation says that the first third is normal songs, last third is podcasts and middle third is a mix (e.g. rap)

CarlinWilliamson commented on src/components/FeatureSelector/Defaults.tsx:
```diff
@@ -62,7 +68,8 @@ const options: FeatureSliderData[] = [
     currentMax: 100,
-    units: '%'
+    units: '%',
+    description: 'Valence describes the musical positiveness conveyed by a track'
```
```suggestion
    description: 'Valence describes the musical/emotional positiveness conveyed by a track'
```
Our less musically inclined users might not understand what musical positiveness is

CarlinWilliamson commented on src/components/FeatureSelector/Defaults.tsx:
```diff
@@ -71,7 +78,8 @@ const options: FeatureSliderData[] = [
     currentMax: 0,
-    units: 'dB'
+    units: 'dB',
+    description : 'The overall loudness of a track in decibels'
```
```suggestion
    description : 'The average loudness of a track in decibels'
```

CarlinWilliamson reviewed (changes requested):
See comments :)

zacharygilmour17 committed:
Update src/components/FeatureSelector/Defaults.tsx

Co-authored-by: CarlinWilliamson <31083798+CarlinWilliamson@users.noreply.github.com>

zacharygilmour17 committed:
Update src/components/FeatureSelector/Defaults.tsx

Co-authored-by: CarlinWilliamson <31083798+CarlinWilliamson@users.noreply.github.com>

zacharygilmour17 committed:
Update src/components/FeatureSelector/Defaults.tsx

Co-authored-by: CarlinWilliamson <31083798+CarlinWilliamson@users.noreply.github.com>

zacharygilmour17 committed:
Update src/components/FeatureSelector/Defaults.tsx

Co-authored-by: CarlinWilliamson <31083798+CarlinWilliamson@users.noreply.github.com>

zacharygilmour17 committed:
removed info icon

zacharygilmour17 committed:
Merge branch 'audio_feature_icon' of https://github.com/Splitify/splitify into audio_feature_icon

CarlinWilliamson reviewed (approved):
LGTM
